src/TLuaInterpreter.cpp

```diff
@@ -12029,6 +12029,41 @@ bool TLuaInterpreter::compile(const QString & code, QString & errorMsg )
     }
 
     int error = luaL_dostring( L, code.toLatin1().data() );
+
+    QString n;
+    if( error != 0 )
+    {
+        string e = "Lua syntax error:";
+        if( lua_isstring( L, 1 ) )
+        {
+            e.append( lua_tostring( L, 1 ) );
+        }
+        errorMsg = "<b><font color='blue'>";
+        errorMsg.append( e.c_str() );
+        errorMsg.append("</b></font>");
+        if( mudlet::debugMode ){ TDebug(QColor(Qt::white),QColor(Qt::red))<<"\n "<<e.c_str()<<"\n">>0;}
+    }
+    else
+    {
+        if( mudlet::debugMode ){ TDebug(QColor(Qt::white),QColor(Qt::darkGreen))<<"\nLUA: code compiled without errors. OK\n" >> 0;}
+    }
+    lua_pop( L, lua_gettop( L ) );
+
+    if( error == 0 ) return true;
+    else return false;
+}
+
+bool TLuaInterpreter::compile(const QString & code, QString & errorMsg, const QString & name)
```

Comment:
I am surprised that the `TLuaInterpreter::compile(...)` allowed the value held in the `TTimer::mName` member to be changed - the `const` in the former's third parameter ought to have prevented modififcation to have take place and ought to have thrown a wobbly at compile time...

Actually, reading up on "getter" functions like `TTimer::getName()` suggests that the declaration in the `TTimer.h` header file should read:

    QString &        getName() const { return mName; }

where the additional `const` means that this function is NOT permitted to change any member of the `TTimer` class (unless that member is also marked with `mutable` apparently)! :astonished: Maybe the absence of that was what allowed it to compile even when you tried to use it as an argument to `bool TLuaInterpreter::compile(const QString & code, QString & errorMsg, const QString & name)`?

Anyhow, I'd tag the timer (or other items) with the appropriate "Trigger:" / "Alias:" / "Key:" etc. at the time it is compiled - so that it never gets stored permanently but can be reported on error.  Otherwise you will probably have to detect whether it is there each time and only put it on if it is absent.  Also note that when we try to go for ___I18n___ with Mudlet 4.0 ( :pray: ) we will want to use the current translated form of "Trigger:" etc. anyhow and if the user changes their language choice there might be transitory issues with pre-change old language terms not being detected by code that looks for the post-change text...